lib/scenic/adapters/postgres/indexes.rb

```diff
@@ -0,0 +1,52 @@
+module Scenic
+  module Adapters
+    class Postgres
+      # Fetches indexes on objects from the Postgres connection.
+      #
+      # @api private
+      class Indexes
+        def initialize(connection:)
+          @connection = connection
+        end
+
+        # Indexes on the provided object.
+        #
+        # @param name [String] The name of the object we want indexes from.
+        # @return [Array<Scenic::Index>]
+        def on(name)
+          indexes_on(name).map(&method(:index_from_database))
+        end
+
+        private
+
+        attr_reader :connection
+
+        def indexes_on(name)
+          connection.execute(<<-SQL)
```

Comment:
If only there was a tool that made it easy to store and execute arbitrary complex queries on a database.